crowbar_framework/lib/crowbar/checks.rb

```diff
@@ -0,0 +1,171 @@
+          :network_json_valid?,
+          :firewall_disabled?,
+          :ping_succeeds?
+        ].each do |check|
+          return false unless send(check)
+        end
+
+        true
+      end
+
+      def fqdn_detected?
+        if hostname.blank? || domain.blank? || fqdn.blank?
+          logger.error("Unable to detect fully-qualified hostname.")
+          return false
+        end
+
+        true
+      end
+
+      def ip_address_resolved?
+        if ipv4_addrs.empty? && ipv6_addrs.empty?
+          logger.error("Could not resolve #{fqdn} to an IPv4 or IPv6 address.")
+          return false
+        end
+
+        true
+      end
+
+      def loopback_resolved?
+        if ipv4_addrs.find { |e| /^127/ =~ e } || ipv6_addrs.find { |e| /^::[0-9]$/ =~ e }
+          logger.error("#{fqdn} resolves to a loopback address.")
+          return false
+        end
+
+        true
+      end
+
+      def ip_configured?
+        ipv4_configured = false
+        ipv6_configured = false
+
+        addr_infos = Socket.ip_address_list
+        addr_infos.each do |addr_info|
+          ipv4_configured = true if ipv4_addrs.include?(addr_info.ip_address)
+          ipv6_configured = true if ipv6_addrs.include?(addr_info.ip_address)
+        end
+
+        unless ipv4_configured || ipv4_addrs.empty?
+          logger.error("No local interface is configured with one of these addresses #{ipv4_addrs.join(" ")}.")
+          return false
+        end
+
+        unless ipv6_configured || ipv6_addrs.empty?
+          logger.error("No local interface is configured with one of these addresses #{ipv6_addrs.join(" ")}.")
+          return false
+        end
+
+        true
+      end
+
+      def network_json_valid?
+        if File.exist?("/etc/crowbar/network.json")
+          network_json = "/etc/crowbar/network.json"
+        else
+          network_json = "/opt/dell/chef/data_bags/crowbar/template-network.json"
+        end
+
+        unless system("network-json-validator --admin-ip \"#{ipv4_addrs.first}\" #{network_json}")
+          logger.error("Failed to validate network.json configuration. Please check and fix with yast2 crowbar.")
+          return false
+        end
+
+        true
+      end
+
+      def firewall_disabled?
+        if system("LANG=C iptables -n -L | grep -qvE '^$ | ^Chain [^ ] | ^target     prot'")
+          logger.error("Firewall is not completely disabled.")
+          return false
+        end
+
+        true
+      end
+
+      def ping_succeeds?
+        `ping -q -c 1 #{fqdn}`
+        unless $?.exitstatus == 0
+          logger.error("Failed to ping #{fqdn}; please check your network configuration.")
+          return false
+        end
+
+        true
+      end
+
+      protected
+
+      def hostname
+        @hostname ||= `hostname`.gsub("\n", "")
+      end
+
+      def domain
+        @domain ||= `hostname -d`.gsub("\n", "")
+      end
+
+      def fqdn
+        @fqdn ||= `hostname -f`.gsub("\n", "")
```

Comment:
Performance/StringReplacement: Use `delete` instead of `gsub`. (https://github.com/JuanitoFatas/fast-ruby#stringgsub-vs-stringtr-code)